task/02-numbers-tasks.js

```diff
@@ -228,7 +228,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    return Number(value) || def;
+    return (value instanceof Number || +value === parseInt(value)) ? value : def;
```

Comment:
1st: should be return {number} type
2nd: see https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/NaN